pkg/controller/workspace/provision/pvc.go

```diff
@@ -0,0 +1,72 @@
+package provision
+
+import (
+	"github.com/che-incubator/che-workspace-operator/pkg/apis/workspace/v1alpha1"
+	"github.com/che-incubator/che-workspace-operator/pkg/config"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func SyncPVC(workspace *v1alpha1.Workspace, components []v1alpha1.ComponentDescription, client client.Client, reqLogger logr.Logger) (err error) {
+	if !IsPVCRequired(components) {
+		return nil
+	}
+
+	var pvc *corev1.PersistentVolumeClaim
+	if pvc, err = generatePVC(workspace, components); err != nil {
+		return err
+	}
+
+	if err := SyncObject(pvc, client, reqLogger); err != nil {
+		return err
+	}
+	return nil
+}
+
+func generatePVC(workspace *v1alpha1.Workspace, components []v1alpha1.ComponentDescription) (*corev1.PersistentVolumeClaim, error) {
+	if !IsPVCRequired(components) {
```

Comment:
it seems a bit redundant since a caller should/and already make sure that PVC is required before generating it.